scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -48,11 +45,8 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)]) {
 
   private[this] val context: ScioContext = self.context
 
-  private[this] val toKvTransform =
-    ParDo.of(Functions.mapFn((kv: (K, V)) => KV.of(kv._1, kv._2)))
-
   private[scio] def toKV(implicit koder: Coder[K], voder: Coder[V]): SCollection[KV[K, V]] =
-    self.applyTransform(toKvTransform).setCoder(CoderMaterializer.kvCoder[K, V](context))
+    self.map(kv => KV.of(kv._1, kv._2)).setCoder(CoderMaterializer.kvCoder[K, V](context))
```

Comment:
`kvCoder` should already be set by `map` so it's either unnecessary or need to be passed to `map` directly to avoid setting a `Coder` twice.

Reply:
yeah I noticed this one but it cant be changed right now. There are some ops that depend on this method and requires the coder to be explicitly `KvCoder`. If we set it through `map` it will still end up as `NullableCoder(KvCoder(NullableCoder(StringUtf8Coder),NullableCoder(StringUtf8Coder)))` which doesn't fly with `GroupByKey`.

I plan to look at this in a separate PR.

Reply:
Hi again @jto! I was thinking, could this fit the bill? #3164 ... there are cases like this where we want the beam coder only and not even wrapped in the `NullCoder` (if the case) coz it doesn't make sense.